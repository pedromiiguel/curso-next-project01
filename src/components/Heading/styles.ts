import styled, { css } from 'styled-components';

export type HeadingProps = {
  colorDark?: boolean;
  size?: 'small' | 'medium' | 'big' | 'huge';
  uppercase?: boolean;
};

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.darkBg : theme.colors.lightBg};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
  `}
`;
