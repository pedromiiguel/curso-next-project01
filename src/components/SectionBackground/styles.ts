import styled, { css } from 'styled-components';

type ContainerBackground = {
  background: boolean;
};

export const Container = styled.div<ContainerBackground>`
  ${({ theme, background }) => css`
    background: ${background ? theme.colors.primaryColor : theme.colors.white};
    color: ${background ? theme.colors.white : theme.colors.primaryColor};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
