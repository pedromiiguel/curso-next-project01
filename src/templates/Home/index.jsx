/* eslint-disable no-unused-vars */

import { Base } from '../Base';
import P from 'prop-types';
import Head from 'next/head';

import { GridTwoColumns } from '../../components/GridTwoColumns';
import { GridContent } from '../../components/GridContent';
import { GridSection } from '../../components/GridSection';
import { GridImage } from '../../components/GridImage';

function Home({ data }) {
  const { menu, sections, footerHtml, slug, title } = data[0];
  const { links, text, link, srcImg } = menu;

  return (
    <>
      <Head>
        <title>{title}</title>
      </Head>
      <Base
        links={links}
        footerHtml={footerHtml}
        logoData={{ text, link, srcImg }}
      >
        {sections.map((section, index) => {
          const { component } = section;
          const key = `${slug}-${index}`;

          if (component === 'section.section-two-columns') {
            return <GridTwoColumns key={key} {...section} />;
          }

          if (component === 'section.section-content') {
            return <GridContent key={key} {...section} />;
          }

          if (component === 'section.section-grid-text') {
            return <GridSection key={key} {...section} />;
          }

          if (component === 'section.section-grid-image') {
            return <GridImage key={key} {...section} />;
          }
        })}
      </Base>
    </>
  );
}

export default Home;

Home.propTypes = {
  data: P.array,
};
