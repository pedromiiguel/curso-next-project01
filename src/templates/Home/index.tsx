/* eslint-disable no-unused-vars */

import { Base } from '../Base';
import Head from 'next/head';

import {
  GridTwoColumns,
  GridTwoColumnsProps,
} from '../../components/GridTwoColumns';
import { GridContent, GridContentProps } from '../../components/GridContent';
import { GridSection, GridSectionProps } from '../../components/GridSection';
import { GridImage, GridImageProps } from '../../components/GridImage';
import { MenuLinkProps } from '../../components/MenuLink';
import { LogoLinkProps } from '../../components/LogoLink';

export type SectionProps =
  | GridContentProps
  | GridImageProps
  | GridSectionProps
  | GridTwoColumnsProps;

export type PageData = {
  title: string;
  slug: string;
  footerHtml: string;
  menu: LogoLinkProps & { links: MenuLinkProps[] };
  sections: SectionProps[];
};

export type HomeProps = {
  data: PageData[];
};

function Home({ data }: HomeProps) {
  const { menu, sections, footerHtml, slug, title } = data[0];
  const { links, text, link, srcImg } = menu;

  return (
    <>
      <Head>
        <title>{title}</title>
        <meta name="description" content="Landing page next" />
      </Head>

      <Base
        links={links}
        footerHtml={footerHtml}
        logoData={{ text, link, srcImg }}
      >
        {sections.map((section, index) => {
          const { component } = section;
          const key = `${slug}-${index}`;

          if (component === 'section.section-two-columns') {
            return (
              <GridTwoColumns key={key} {...(section as GridTwoColumnsProps)} />
            );
          }

          if (component === 'section.section-content') {
            return <GridContent key={key} {...(section as GridContentProps)} />;
          }

          if (component === 'section.section-grid-text') {
            return <GridSection key={key} {...(section as GridSectionProps)} />;
          }

          if (component === 'section.section-grid-image') {
            return <GridImage key={key} {...(section as GridImageProps)} />;
          }
        })}
      </Base>
    </>
  );
}

export default Home;
