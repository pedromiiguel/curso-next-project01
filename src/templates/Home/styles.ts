import styled, { css } from 'styled-components';

type HomeBackground = {
  background: string;
};

export const Wrapper = styled.div<HomeBackground>`
  ${({ background }) => css`
    background: ${background};
  `}
`;
