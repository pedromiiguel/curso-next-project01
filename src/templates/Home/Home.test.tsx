import { renderTheme } from '../../styles/render-theme';
import Home from '.';

describe('<Home />', () => {
  it('should render', () => {
    const { container } = renderTheme(
      <Home
        data={[
          {
            menu: { links: [], text: '', link: '', srcImg: '' },
            sections: [],
            footerHtml: '',
            slug: '',
            title: '',
          },
        ]}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
