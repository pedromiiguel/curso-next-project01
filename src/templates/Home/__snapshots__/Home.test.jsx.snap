// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c3 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c0 {
  background: #ffffff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Error 404
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              A página que você busca não foi encontrada. 
              <a
                href="/"
              >
                Clique aqui para voltar
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
